### PR TITLE
Fix S1854 FP: Value used in finally should LiveIn for all try blocks

### LINKED ISSUE #9440: Fix S1854 FP: Value used in finally should LiveIn for all try blocks
Should not raise in complex scenarios related to Roslyn LVA:

```C#
public void UsedInFinally()
{
            int value = 42; // Noncompliant FP
            try
            {
                SomethingThatCanThrow();
                value = 0;
            }
            finally
            {
                Use(value);
            }
}
```

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #9440

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs [hunk 1]
```diff
@@ -192,7 +192,7 @@ void calculateRate2(int a, int b)
                 x = 11; // FN, muted due to try/catch
                 x = 12;
                 Console.Write(x);
-                x = 13; // Compliant, Console.Write could throw and value is read fater try/catch
+                x = 13; // FN, muted due to try/catch
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:195]: I'd add a more explanative comment here (and later) - mentioning the change in LVA.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs [hunk 2]
```diff
@@ -1436,7 +1436,7 @@ public void UsedInFinally_NestedInLambda()
             {
             }
         }
-
+
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1439]: 👾 👾 👾

Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1439]: (space invader)

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs (hunk 1 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:195]: Not needed -you are fixing this later.

File: analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.Test/TestCases/DeadStores.RoslynCfg.cs:1439]: Will fix in the next PR

**Pull request merged**